sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -73,6 +74,34 @@ public BinaryData(string data)
             Bytes = s_encoding.GetBytes(data);
         }
 
+        /// <summary>
+        /// Creates a binary data instance from an object and serializes it
+        /// using the <see cref="JsonObjectSerializer"/>.
+        /// </summary>
+        /// <param name="data">The data that will be serialized.</param>
+        /// <param name="type">The type of the data.</param>
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        public BinaryData(object data, Type type) :
```

Comment:
This doesn't at all look like something that would do JsonSerialization from the outside.

Reply:
This is harder to use than the generic overload and clutters a valuable space of public ctors. Why do we think it's worth it?

Reply:
The main reason would be discoverability, but I agree that it is somewhat of a high price to pay in that we now have two ways of doing the same thing.
/cc @KrzysztofCwalina